nyx_client/data.py

```diff
@@ -17,6 +17,7 @@
 import logging
 
 import requests
+from typing_extensions import deprecated
```

Comment:
Either:
1. Explicitly specify `typing_extensions` as a dependency **or**
2. Replace with a call to `warnings.warn("...", DeprecationWarning)` (since that's all it does anyway)

Personally I'd go with (2) since e.g. `mypy` doesn't [support typing deprecation checks yet anyway](https://mypy.readthedocs.io/en/stable/search.html?q=deprecated&check_keywords=yes&area=default#)